cmake/pcl_find_sse.cmake

```diff
@@ -223,6 +217,15 @@ function(PCL_CHECK_FOR_SSE)
       # platforms.
       list(APPEND SSE_FLAGS "-ffloat-store")
     endif()
+    
+    if(PCL_ENABLE_MARCHNATIVE AND (NOT CMAKE_CROSSCOMPILING))
```

Comment:
Can we remove this and only depend on the condition with HAVE_MARCH?

Reply:
Seems to be possible, but I would say it is better (for safety and for understanding) to check everything: The flags should only be added if the user wants them (`PCL_ENABLE_MARCHNATIVE`), if we are not crosscompiling (`NOT CMAKE_CROSSCOMPILING`), _and_ if the compiler knows the flag (`HAVE_MARCH`)

Reply:
L11 ensures that HAVE_MARCH is enabled only if this exact condition is valid.

Reply:
But when would we set the `-mtune=native` flag? Only checking that `HAVE_MARCH==false` does not seem to be sufficient since `CMAKE_CROSSCOMPILING` could be true or `PCL_ENABLE_MARCHNATIVE` could be false/off (I would interpret it so that if `PCL_ENABLE_MARCHNATIVE` is false/off, neither `march=native` nor `mtune=native` is set)